task/thermal/src/control.rs

```diff
@@ -77,6 +77,42 @@ impl TemperatureSensor {
     }
 }
 
+/// Represents the indvidual fans in the system
+///
+/// Depending on the system we have diferent numbers of fans structured in
+/// different ways. Not all fans are guaranteed to be there at all times so
+/// their corresponding sensor is an `Option`. We should not read the RPM of
+/// fans which are not present and their PWM should only be driven low.
+#[derive(Copy, Clone)]
+pub struct Fans([Option<SensorId>; bsp::NUM_FANS]);
+
+impl Fans {
+    pub fn new() -> Self {
+        Self([None; bsp::NUM_FANS])
+    }
+    pub fn add(&mut self, index: usize, sensor: SensorId) {
```

Comment:
I'd be tempted to define `core::ops::Index/IndexMut` instead of `add/remove`, because they've got less ambiguous semantics (e.g. I was wondering whether `add(0, ...)` shifts items).

Reply:
I think that is an excellent idea.